pulser-core/pulser/backend/observable.py

```diff
@@ -99,6 +99,16 @@ def __init__(
     def _base_tag(self) -> str:
         pass
 
+    def _expected_kwargs(self) -> set[str]:
+        return {"evaluation_times", "_tag_suffix"}
+
+    def _to_abstract_repr(self) -> dict[str, Any]:
+        obs_kwargs = {
+            key: self.__dict__[key] for key in self._expected_kwargs()
+        }
+        obs_tag = self._base_tag
+        return {obs_tag: obs_kwargs}
```

Comment:
I must say I find this unnecessarily convoluted and restrictive. I'd imagine in your schema you might be using `"tag_suffix"` instead of `"_tag_suffix"`, for instance. You could just do this:
```suggestion
    def _to_abstract_repr(self) -> dict[str, Any]:
        return {self._base_tag: {
            "evaluation_times": self.evaluation_times,
            "tag_suffix": self._tag_suffix,
        }
        }
```
This is simpler, more explicit and more flexible. For the subclasses, you can simply do e.g.
```
    def _to_abstract_repr(self) -> dict[str, Any]:
        repr = super()._to_abstract_repr()
        repr[self._base_tag]["one_state"]: self.one_state,
        return repr
```
I think this is not so bad

Reply:
done